CMakeLists.txt

```diff
@@ -89,8 +108,11 @@ set(SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 #TODO set(DOC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/doc)
 set(TEST_DIR ${CMAKE_CURRENT_SOURCE_DIR}/tests)
 
-# Enable all subprojects to find each other's include files
-include_directories(${SRC_DIR})
+include_directories(
+    ${SRC_DIR} # Enable all subprojects to find each other's include files
```

Comment:
What is the rationale behind this? In modern CMake, we work with remote targets. The include directories are then implictly added when the respective remote target is listed in the in target_link_libraries.
See, e.g., https://github.com/bmwcarit/MoCOCrW/blob/openssl1.1/src/CMakeLists.txt#L70

Anything that is broken for you in particular?

Reply:
Unfortunately, this is more of a work-around to a known problem with finding OpenSSL libraries. Basically, building a recent version of OpenSSL (in my case > 3.00), at a custom path, results in installation to place the library inside a directory called "lib64". Despite setting OPENSSL_ROOT_DIR, `find_package` does not find the library:

```
  Could NOT find OpenSSL, try to set the path to OpenSSL root folder in the
  system variable OPENSSL_ROOT_DIR (missing: OPENSSL_CRYPTO_LIBRARY)
```

As an initial work-around, I manually set OPENSSL variables as parameters so that cmake is able to find the library and headers (hence the rational behind `include_directories`). However, thinking about this again, I don't think this is the right approach either. 

Turns out that a similar issue was encountered by CMake devs (https://gitlab.kitware.com/cmake/cmake/-/issues/22945) with FindOpenSSL. I built and installed the current version of CMake, namely 3.23.1, and cmake managed to find the library inside "lib64". On CMake version 3.16.3, it does not work for me. 

Therefore, at some point, we should add `cmake_minimum_required` so that MoCOCrW users can easily integrate custom OpenSSL builds. I think I should remove that change; users can simply rename lib64 to lib in the meantime, if preferred over changing the CMakeLists.txt